alloy-core/src/main/java/org/starchartlabs/alloy/core/Preconditions.java

```diff
@@ -81,6 +82,29 @@ public static void checkArgument(boolean expression) {
         }
     }
 
```

Comment:
Implementation-wise, this look good, but I'd update the documentation to the follow:

```
/**
     * Evaluates a value against a provided {@link Predicate} expression
     *
     * @param value
     *            An assignable value to evaluate
     * @param predicate
     *            An operation to evaluate against {@code value}
     * @param <T>
     *            The type of the value being evaluated
     * @return {@code value} if the provided {@code predicate} evaluates to true
     * @throws IllegalArgumentException
     *             If {@code predicate} evaluates to false
     * @since 0.2.0
     */
```

Reply:
Ok docs updated